iOS Client/BookmarksController/MBBookmarksViewController.swift

```diff
@@ -18,24 +26,122 @@ class MBBookmarksViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        // Do any additional setup after loading the view.
+        self.title = "Bookmarks"
+        self.isEditing = false
+        self.navigationItem.leftBarButtonItem = editButtonItem
+        
+        let fetchRequest: NSFetchRequest<MBArticle> = MBArticle.fetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "isBookmarked == %@", NSNumber(value: true))
+        
+        let sort = NSSortDescriptor(key: #keyPath(MBArticle.date), ascending: false)
+        fetchRequest.sortDescriptors = [sort]
+        
+        fetchedResultsController = NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: managedObjectContext, sectionNameKeyPath: nil, cacheName: nil)
+        
+        fetchedResultsController.delegate = self
+        
+        do {
+            try fetchedResultsController.performFetch()
+        } catch let error as NSError {
+            print("Fetching error: \(error), \(error.userInfo)")
+        }
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
-    
+}
 
-    /*
-    // MARK: - Navigation
+// MARK: - Internal
+extension MBBookmarksViewController {
+    func configure(cell: UITableViewCell, for indexPath: IndexPath) {
+        guard let cell = cell as? BookmarkCell else {
+            return
+        }
+        
+        let article = fetchedResultsController.object(at: indexPath)
+        cell.titleLabel.attributedText = article.title?.convertHtml()
+        cell.titleLabel.font = UIFont.preferredFont(forTextStyle: .subheadline)
+        cell.titleLabel.textColor = UIColor.ArticleTitle
+        cell.titleLabel.sizeToFit()
+        
+        if let savedData = article.image?.image {
+            cell.coverImageView.image = UIImage(data: savedData as Data)
+        }
+    }
+}
 
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destinationViewController.
-        // Pass the selected object to the new view controller.
+// MARK: - NSFetchedResultsControllerDelegate
+extension MBBookmarksViewController: NSFetchedResultsControllerDelegate {
+    func controllerWillChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
+        tableView.beginUpdates()
     }
-    */
+    
+    func controller(_ controller: NSFetchedResultsController<NSFetchRequestResult>, didChange anObject: Any, at indexPath: IndexPath?, for type: NSFetchedResultsChangeType, newIndexPath: IndexPath?) {
```

Comment:
I'll be honest not sure how this FetchResultsController works so I'm going to trust you got this one

Reply:
ah yeah dude, I copied all this from the book